pylib/cqlshlib/cloud.py

```diff
@@ -0,0 +1,74 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import random
+from cassandra.cluster import Cluster
+from cassandra.policies import WhiteListRoundRobinPolicy, RoundRobinPolicy
+from cassandra.pool import HostDistance
+
+
+class EndpointWhiteListRoundRobinPolicy(WhiteListRoundRobinPolicy):
+    def __init__(self, hosts):
+        self._allowed_hosts = self._allowed_hosts_resolved = tuple(hosts)
+        RoundRobinPolicy.__init__(self)
```

Comment:
I have no idea why it is `RoundRobinPolicy` and not `super()`. 
This is how it was in the original code and I think it is kind of ... incorrect?

Reply:
Looking at [WhiteListRoundRobinPolicy's constructor](https://github.com/datastax/python-driver/blob/15d715f4e686032b02ce785eca1d176d2b25e32b/cassandra/policies.py#L419-L428), I think that the reason for `RoundRobinPolicy.__init__(self)` rather than `super()...` is to override `_allowed_hosts` and `_allowed_hosts_resolved`.

I agree with you that it is very confusing. WDYT about adding a brief comment explaining the intention?